src/SharpLearning.Optimization/BayesianOptimizer.cs

```diff
@@ -188,36 +214,38 @@ public OptimizerResult[] Optimize(Func<double[], OptimizerResult> functionToMini
         {
             var bestParameterSet = new double[m_parameters.Length];
             var bestParameterSetScore = double.MaxValue;
-
-            var parameterSets = new List<double[]>();
-            var parameterSetScores = new List<double>();
+            var parameterSets = new ConcurrentBag<double[]>();
+            var parameterSetScores = new ConcurrentBag<double>();
 
             var usePreviousResults = m_previousParameterSetScores != null && m_previousParameterSets != null;
 
             int iterations = 0;
 
             if (usePreviousResults)
             {
-                parameterSets.AddRange(m_previousParameterSets);
-                parameterSetScores.AddRange(m_previousParameterSetScores);
+                for (int i = 0; i < m_previousParameterSets.Count; i++)
+                {
+                    parameterSets.Add(m_previousParameterSets[i]);
+                    parameterSetScores.Add(m_previousParameterSetScores[i]);
+                }
 
-                for (int i = 0; i < parameterSets.Count; i++)
+                Parallel.For(0, parameterSets.Count, m_parallelOptions, i =>
```

Comment:
@jameschch I don't think it makes sense to parallelize this part of the code since it is very light weight, so the overhead of spinning up threads, and the added complexity is not worth it in my opinion.

Reply:
Yes, agree.